src/lib/twizzler-abi/src/thread.rs

```diff
@@ -74,6 +78,65 @@ impl ExecutionState {
     }
 }
 
+
+#[cfg(kani)]
+mod thread_verif{
+    use super::ExecutionState;
+    use super::ThreadRepr;
+    use crate::syscall::Syscall;
+    use crate::arch::syscall::raw_syscall;
+
+    //Not proper how do we mock/stub a syscall
+    pub fn stub_raw_syscall(call: Syscall, args: &[u64]) -> (u64, u64) {
+        unsafe {
+            if core::intrinsics::unlikely(args.len() > 6) {
+                crate::print_err("too many arguments to raw_syscall");
+                crate::internal_abort();
+            }
+            let a0 = *args.first().unwrap_or(&0u64);
+            let a1 = *args.get(1).unwrap_or(&0u64);
+            let mut a2 = *args.get(2).unwrap_or(&0u64);
+            let a3 = *args.get(3).unwrap_or(&0u64);
+            let a4 = *args.get(4).unwrap_or(&0u64);
+            let a5 = *args.get(5).unwrap_or(&0u64);
+        
+            let mut num = call.num();
+            //If we know what system call is being called we can stub that out too
+        
+            // core::arch::asm!("syscall", inout("rax") num, in("rdi") a0, in("rsi") a1, inout("rdx") a2, in("r10") a3, in("r9") a4, in("r8") a5, lateout("rcx") _, lateout("r11") _, clobber_abi("system"));
+            (kani::any(), kani::any())
+        }
+        
+    }
+
+    #[kani::proof]
+    #[kani::stub(crate::arch::syscall::raw_syscall, stub_raw_syscall)]
+    pub fn fixme_thread_set_state(){
+    
+        let tr = ThreadRepr::default();
+        
+        let state = exec_state();
+
+        tr.set_state(state, kani::any());
+        
+        assert!(tr.get_state() == state);
+
+    }
+
```

Comment:
Can't we derive kani::Arbitrary (https://model-checking.github.io/kani/tutorial-nondeterministic-variables.html#custom-nondeterministic-types) for some of these types? I recall doing that, but I'm not the kani expert here :)